src/common/exif.cc

```diff
@@ -3988,8 +4005,31 @@ int dt_exif_xmp_attach_export(const int imgid, const char *filename, void *metad
   }
   catch(Exiv2::AnyError &e)
   {
-    std::cerr << "[dt_exif_xmp_attach_export] " << filename << ": caught exiv2 exception '" << e << "'\n";
-    return -1;
+    std::string s(e.what());
+    if(s == "Size of XMP JPEG segment is larger than 65535 bytes")
+    {
+      Exiv2::XmpData &xmpData = img->xmpData();
+      _remove_xmp_keys(xmpData, "Xmp.darktable.history");
+      _remove_xmp_keys(xmpData, "Xmp.darktable.masks_history");
+      _remove_xmp_keys(xmpData, "Xmp.darktable.auto_presets_applied");
+      _remove_xmp_keys(xmpData, "Xmp.darktable.iop_order");
```

Comment:
finally you did not removed iop_order_list, is that expected ?

Reply:
> How to reproduce the issue on #4355 ? I tried the XMP posted there. I have added metadata and when I export I have them in the jpeg. Maybe I'm missing something ?

Ok found, I had not the history exported :)

Reply:
In fact, in the four strings above represents the needles => more 1800 records in case of the test file, and iop_order_list is included.